front/src/modules/timetableItem/components/Timetable/types.ts

```diff
@@ -20,30 +20,32 @@ export type TrainCategoryFilter = 'all' | 'noCategory' | TrainMainCategory;
 
 type SimulationSummaryResultSuccess = Extract<SimulationSummaryResult, { status: 'success' }>;
 
-type TimetableItemWithSummaries = Omit<
+export type SimulationSummary =
+  | { isValid: false; invalidReason: InvalidReason }
+  | {
+      isValid: true;
+      duration: Duration /** Travel time */;
+      pathLength: string;
+      mechanicalEnergyConsumed: number;
+      notHonoredReason?: 'scheduleNotHonored' | 'trainTooFast';
+      pathItemTimes: {
+        base: SimulationSummaryResultSuccess['path_item_times_base'];
+        provisional: SimulationSummaryResultSuccess['path_item_times_provisional'];
+        final: SimulationSummaryResultSuccess['path_item_times_final'];
+      };
+    };
+
+export type TimetableItemWithSummaries = Omit<
   TrainScheduleResponse,
   'id' | 'train_name' | 'rolling_stock_name' | 'timetable_id' | 'start_time'
 > & {
   name: string;
   startTime: Date;
-  arrivalTime: Date | null;
-  duration: Duration | null;
   stopsCount: number;
-  pathLength: string;
   rollingStock?: LightRollingStockWithLiveries;
-  mechanicalEnergyConsumed: number;
   speedLimitTag: string | null;
   labels: string[];
-  invalidReason?: InvalidReason;
-  notHonoredReason?: 'scheduleNotHonored' | 'trainTooFast';
-  scheduledPointsNotHonored?: boolean;
-  isValid: boolean;
-  pathItemTimes?: {
-    base: SimulationSummaryResultSuccess['path_item_times_base'];
-    provisional: SimulationSummaryResultSuccess['path_item_times_provisional'];
-    final: SimulationSummaryResultSuccess['path_item_times_final'];
-  };
-};
+} & (({ isLoaded: true } & SimulationSummary) | { isLoaded: false; isValid: undefined });
```

Comment:
Do we have to deal with `isLoaded` everywhere ? Won't it be possible to just use `isValid` ? (if `isValid` is not defined, it means the summaries haven't loaded yet)
```suggestion
} & (({ isValid: boolean } & SimulationSummary) | { isValid: undefined });
```

Reply:
Agreed. "isLoaded" is also a bit vague (what is loading? the train details, projection, or simulation?)

Reply:
I modified the type, is it better now ?

Reply:
Works for me :)